// nylas/lib/models/event-reminder-method.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var attributes_1 = __importDefault(require("./attributes"));
var model_1 = __importDefault(require("./model"));
var EventReminderMethod;
(function (EventReminderMethod) {
    EventReminderMethod["Email"] = "email";
    EventReminderMethod["Popup"] = "popup";
    EventReminderMethod["Display"] = "display";
    EventReminderMethod["Sound"] = "sound";
})(EventReminderMethod = exports.EventReminderMethod || (exports.EventReminderMethod = {}));
var EventReminder = /** @class */ (function (_super) {
    __extends(EventReminder, _super);
    function EventReminder(props) {
        var _this = _super.call(this) || this;
        _this.initAttributes(props);
        return _this;
    }
    EventReminder.attributes = {
        reminderMinutes: attributes_1.default.String({
            modelKey: 'reminderMinutes',
            jsonKey: 'reminder_minutes',
        }),
        reminderMethod: attributes_1.default.String({
            modelKey: 'reminderMethod',
            jsonKey: 'reminder_method',
        }),
    };
    return EventReminder;
}(model_1.default));
exports.EventReminder = EventReminder;
